src/main/java/uk/gov/hmcts/reform/cpo/security/IdamRepository.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.reform.cpo.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+@Component
+public class IdamRepository {
+
+    private final IdamClient idamClient;
+
+    @Autowired
+    public IdamRepository(IdamClient idamClient) {
+        this.idamClient = idamClient;
+    }
+
+    @Cacheable("userInfoCache")
+    public UserInfo getUserInfo(String bearerToken) {
```

Comment:
Could we possibly add a `SecurityUtils` class as part of this PR too please? Not required but would be useful for future :) Perhaps for now just with the (using ACA as an example) `getUserInfo()` and `getServiceNameFromS2SToken()` methods, as I think they'll be needed for various things.